src/data/transforms.py

```diff
@@ -1,169 +1,233 @@
-        for t in self.transforms:
-            format_string += '\n'
-            format_string += '    {0}'.format(t)
-        format_string += '\n)'
-        return format_string
+        return self.__class__.__name__
 
 
 class ToTensor(BaseTransform):
     """Convert a tuple of numpy.ndarray to a tuple of torch.Tensor.
+    Default is to transform a tuple of numpy.ndarray to a tuple of channel-first torch.Tensor
+    which infer data types from the tuple of numpy.ndarray.
+
+    Args:
+        channel_first (bool, optional): Whether the data format of the output data is channel-first,
+            that is (H, W, C) to (C, H, W) and (H, W, D, C) to (C, D, H, W) (default: True).
     """
 
-    def __init__(self):
+    def __init__(self, channel_first=True):
         super().__init__()
+        self.channel_first = channel_first
 
-    def __call__(self, *imgs, dtypes=None, **kwargs):
+    def __call__(self, *imgs, dtypes=None):
         """
         Args:
-            imgs (tuple of numpy.ndarray): The images to be converted to tensor.
-            dtypes (sequence of torch.dtype, optional): The corresponding dtype of the images
-                (default: None, transform all the images' dtype to torch.float).
+            imgs (tuple of numpy.ndarray): The images to be converted.
+            dtypes (sequence of torch.dtype, optional): The dtypes of the converted images
+                (default: None, infer data types from the images).
 
         Returns:
             imgs (tuple of torch.Tensor): The converted images.
         """
         if not all(isinstance(img, np.ndarray) for img in imgs):
             raise TypeError('All of the images should be numpy.ndarray.')
-
-        if dtypes:
-            if not all(isinstance(dtype, torch.dtype) for dtype in dtypes):
-                raise TypeError('All of the dtypes should be torch.dtype.')
-            if len(dtypes) != len(imgs):
-                raise ValueError('The number of the dtypes should be the same as the images.')
-            imgs = tuple(img.to(dtype) for img, dtype in zip(map(torch.as_tensor, imgs), dtypes))
-        else:
-            imgs = tuple(img.float() for img in map(torch.as_tensor, imgs))
+        if not all(img.ndim == 3 for img in imgs) and not all(img.ndim == 4 for img in imgs):
+            raise ValueError('All of the images should be 2D or 3D with channels.')
+
+        if dtypes is None:
+            dtypes = tuple(None for _ in range(len(imgs)))
+        if len(dtypes) != len(imgs):
+            raise ValueError('The number of the dtypes should be the same as the images.')
+        imgs = tuple(torch.as_tensor(img, dtype=dtype) for img, dtype in zip(imgs, dtypes))
+        if self.channel_first:
+            if imgs[0].ndim == 3:
+                imgs = tuple(img.permute(2, 0, 1).contiguous() for img in imgs)
+            elif imgs[0].ndim == 4:
+                imgs = tuple(img.permute(3, 2, 0, 1).contiguous() for img in imgs)
```

Comment:
`elif self.channel_first == False` ...

Reply:
if `self.channel_first is False`, we do not need to do anything here.

Reply:
confirmed.